File: src/main/java/nl/vollo/kern/security/CookieService.java
package nl.vollo.kern.security;

import org.springframework.stereotype.Service;

import javax.servlet.http.Cookie;
import javax.servlet.http.HttpServletRequest;

@Service
public class CookieService {

    public static final String VOLLO_TOKEN_NAME = "vollo_token";

    public Cookie nieuweCookie(HttpServletRequest request, String token) {
        Cookie cookie = new Cookie(VOLLO_TOKEN_NAME, token);
        cookie.setHttpOnly(true);
        cookie.setSecure(request.isSecure());
        cookie.setPath("/");
        cookie.setMaxAge(-1);
        return cookie;
    }

    public Cookie verwijderCookie(HttpServletRequest request) {
        Cookie cookie = new Cookie(VOLLO_TOKEN_NAME, "");
        cookie.setHttpOnly(true);
        cookie.setSecure(request.isSecure());
        cookie.setPath("/");
        cookie.setMaxAge(0);
        return cookie;
    }

}

File: src/main/kotlin/nl/vollo/kern/security/CookieService.kt
package nl.vollo.kern.security

import org.springframework.stereotype.Service
import javax.servlet.http.Cookie
import javax.servlet.http.HttpServletRequest

@Service
class CookieService {

    fun nieuweCookie(request: HttpServletRequest, token: String): Cookie {
        val cookie: Cookie = Cookie(VOLLO_TOKEN_NAME, token)
        cookie.isHttpOnly = true
        cookie.secure = request.isSecure
        cookie.path = "/"
        cookie.maxAge = -1
        return cookie;
    }

    fun verwijderCookie(request: HttpServletRequest): Cookie {
        val cookie: Cookie = Cookie(VOLLO_TOKEN_NAME, "")
        cookie.isHttpOnly = true
        cookie.secure = request.isSecure
        cookie.path = "/"
        cookie.maxAge = 0
        return cookie;
    }

    companion object {
        const val VOLLO_TOKEN_NAME = "vollo_token"
    }

}
